app/src/main/java/com/vultisig/wallet/ui/models/VaultAccountsViewModel.kt

```diff
@@ -42,51 +41,78 @@ internal data class AccountUiModel(
 
 @HiltViewModel
 internal class VaultAccountsViewModel @Inject constructor(
+    private val navigator: Navigator<Destination>,
+
     private val addressToUiModelMapper: AddressToUiModelMapper,
     private val fiatValueToStringMapper: FiatValueToStringMapper,
 
     private val vaultRepository: VaultRepository,
     private val accountsRepository: AccountsRepository,
 ) : ViewModel() {
-    var vault: Vault? by mutableStateOf(null)
+    private var vaultId: String? = null
 
     val uiState = MutableStateFlow(VaultAccountsUiModel())
 
     private var loadVaultNameJob: Job? = null
     private var loadAccountsJob: Job? = null
 
     fun loadData(vaultId: String) {
+        this.vaultId = vaultId
         loadVaultName(vaultId)
-        loadAccounts(vaultId)
+        loadAccounts(
+            vaultId = vaultId,
+            showRefreshing = false,
+        )
```

Comment:
There's changes from my other PR in this one, please do something with git history of this branch